saq/queue/postgres.py

```diff
@@ -108,10 +114,20 @@ def __init__(
     ) -> None:
         super().__init__(name=name, dump=dump, load=load, swept_error_message=swept_error_message)
 
+        if pool is None and url is None:
+            raise ValueError("Either pool or url must be provided")
+        if pool is not None and url is not None:
+            raise ValueError("Either pool or url must be provided, not both")
+
         self.versions_table = Identifier(versions_table)
         self.jobs_table = Identifier(jobs_table)
         self.stats_table = Identifier(stats_table)
-        self.pool = pool
+        self.pool = pool or AsyncConnectionPool(
+            url,  # type: ignore
+            check=AsyncConnectionPool.check_connection,
+            open=False,
+        )
+        self._passed_in_pool = pool is not None
```

Comment:
Perhaps it's just me but I had a hard time understanding that this is actually a bool variable. Should this be something like `is_pool_provided`?

Reply:
changed, yours is better